src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -704,7 +704,8 @@ def _kind_find_or_create(name, symbol_table):
             new_symbol = DataSymbol(lower_name,
                                     default_integer_type(),
                                     visibility=kind_symbol.visibility,
-                                    interface=kind_symbol.interface)
+                                    interface=kind_symbol.interface,
+                                    is_constant=True)
             table.swap(kind_symbol, new_symbol)
```

Comment:
I tried to use specialise in the next two branches but it all began to unravel - `specialise` complains if given a symbol of the same class as the one being specialised and I also hit problems because a bare Symbol doesn't have a datatype attribute. All in all, it was simpler to keep the special case for the `DataSymbol` which just modifies its properties.